packages/dds/tree/src/feature-libraries/sequence-field/sequenceFieldEditor.ts

```diff
@@ -56,31 +74,49 @@ export interface SequenceFieldEditor extends FieldEditor<Changeset> {
 		destIndex: number,
 		detachCellId: CellId,
 		attachCellId: CellId,
+		revision: RevisionTag,
 	): Changeset;
 }
 
 export const sequenceFieldEditor = {
 	buildChildChange: (index: number, change: NodeId): Changeset =>
 		markAtIndex(index, { count: 1, changes: change }),
-	insert: (index: number, count: number, firstId: ChangesetLocalId): Changeset => {
+	insert: (
+		index: number,
+		count: number,
+		firstId: ChangesetLocalId,
+		revision: RevisionTag,
+	): Changeset => {
 		const mark: CellMark<Insert> = {
 			type: "Insert",
 			id: firstId,
 			count,
-			cellId: { localId: firstId },
+			cellId: { localId: firstId, revision },
+			revision,
 		};
 		return markAtIndex(index, mark);
 	},
-	remove: (index: number, count: number, id: ChangesetLocalId): Changeset =>
-		count === 0 ? [] : markAtIndex(index, { type: "Remove", count, id }),
+	remove: (
+		index: number,
+		count: number,
+		id: ChangesetLocalId,
+		revision: RevisionTag,
+	): Changeset =>
+		count === 0 ? [] : markAtIndex(index, { type: "Remove", count, id, revision }),
 
-	revive: (index: number, count: number, detachEvent: CellId): Changeset => {
+	revive: (
+		index: number,
+		count: number,
+		detachEvent: CellId,
+		revision: RevisionTag,
+	): Changeset => {
 		assert(detachEvent.revision !== undefined, 0x724 /* Detach event must have a revision */);
 		const mark: CellMark<Insert> = {
 			type: "Insert",
 			id: detachEvent.localId,
 			count,
-			cellId: detachEvent,
+			cellId: { ...detachEvent, revision },
```

Comment:
I think that overriding the detachEvent.revision is not what we want here: the detach event describes the change that removed the content that is being revived. That's a prior change independent from this one.